erpnext/buying/doctype/purchase_order/purchase_order.js

```diff
@@ -181,6 +181,18 @@ erpnext.buying.PurchaseOrderController = erpnext.buying.BuyingController.extend(
 				})
 			}, __("Add items from"));
 
+		cur_frm.add_custom_button(__('For Supplier'),
+			function() {
+				erpnext.utils.map_current_doc({
```

Comment:
There is already a function for this, just need to bind it to the custom button
https://github.com/frappe/erpnext/blob/develop/erpnext/buying/doctype/purchase_order/purchase_order.js#L94